src/plugins/vis_types/heatmap/public/editor/components/heatmap.tsx

```diff
@@ -79,12 +83,22 @@ const HeatmapOptions = (props: HeatmapOptionsProps) => {
 
         <BasicOptions {...props} legendPositions={legendPositions} />
         {showElasticChartsOptions && (
-          <LongLegendOptions
-            data-test-subj="heatmapLongLegendsOptions"
-            truncateLegend={stateParams.truncateLegend ?? true}
-            maxLegendLines={stateParams.maxLegendLines ?? 1}
-            setValue={setValue}
-          />
+          <>
+            <LongLegendOptions
+              data-test-subj="heatmapLongLegendsOptions"
+              truncateLegend={stateParams.truncateLegend ?? true}
+              maxLegendLines={stateParams.maxLegendLines ?? 1}
+              setValue={setValue}
+            />
+            <LegendSizeSettings
+              legendSize={stateParams.legendSize}
+              onLegendSizeChange={handleLegendSizeChange}
+              isVerticalLegend={
```

Comment:
it seems to me that `isVerticalLegend` is not a very good name, but in Lens we also use it, so we’ll probably leave it as it is...